### PR TITLE
Reversed dependency between CC and backup

### PR DESCRIPTION
**Pull request by phughk**
Before this commit, CC is dependant on backup.
Applying this commit will shuffle dependencies around to be more sensible

### CODE REVIEW

**Comment by phughk** [comment]
@martinfurmanski once you have reviewed this it is ready for merge, I have already prepared the merge commit for 3.3

File: enterprise/cluster/pom.xml [hunk 1]
```diff
@@ -104,5 +104,10 @@
       <artifactId>mockito-core</artifactId>
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>org.neo4j</groupId>
+      <artifactId>neo4j-enterprise-kernel</artifactId>
+      <version>${project.version}</version>
+    </dependency>
```
Comment (andrewkerr9000) [inline enterprise/cluster/pom.xml:111]: Where is this needed?

**Review by andrewkerr9000 — APPROVED** [review]
:+1: Although I'm not sure where the dependencies on neo4j-enterprise-kernel are needed.

File: enterprise/cluster/pom.xml (hunk 1 above)
Comment (phughk) [reply enterprise/cluster/pom.xml:111]: Used for `EnterpriseEditionSettings`

**Comment by digitalstain** [comment]
We cannot merge this until the replacement tool for CC backup is in place, correct?

**Pull request closed without merging**